src/org/exist/security/AbstractRealm.java

```diff
@@ -65,7 +63,7 @@
     protected final PrincipalDbByName<Group> groupsByName = new PrincipalDbByName<>();
     
 
-    private SecurityManager sm;
+    private SecurityManagerImpl sm;
```

Comment:
Why concrete class? Could you not just modify the interface for what seems to be function naming & argument changes?